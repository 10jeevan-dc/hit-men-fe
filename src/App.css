.App {
  text-align: center;
  background-color: #1F3944;
  min-height: 100vh;
}
