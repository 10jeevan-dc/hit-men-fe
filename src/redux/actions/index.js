import { createAction } from 'redux-actions';

// Action constants

export const RESET_USER_DETAILS = 'RESET_USER_DETAILS';
export const SET_USER_ALLOCATION_DETAILS = 'SET_USER_ALLOCATION_DETAILS';
export const RESET_GAME_STATE = 'RESET_GAME_STATE';
export const ALLOW_GAME_PLAY = 'ALLOW_GAME_PLAY';
export const CHANGE_GAME_STATE = 'CHANGE_GAME_STATE';
export const USER_DETAILS_SET = 'USER_DETAILS_SET';
export const SET_USER_SCORE = 'SET_USER_SCORE';
export const SET_REMAINING_TIME = 'SET_REMAINING_TIME';
export const SET_LEADERBOARD_DATA = 'SET_LEADERBOARD_DATA';
export const SET_TEAM_SCORE_DATA = 'SET_TEAM_SCORE_DATA';


// Actions

export const setUserAllocationDetails = createAction(SET_USER_ALLOCATION_DETAILS);
export const resetUserDetails = createAction(RESET_USER_DETAILS);
export const allowGamePlay = createAction(ALLOW_GAME_PLAY);
export const changeGameState = createAction(CHANGE_GAME_STATE);
export const resetGameState = createAction(RESET_GAME_STATE);
export const userDetailsSet = createAction(USER_DETAILS_SET);
export const setUserScore = createAction(SET_USER_SCORE);
export const setRemainingTime = createAction(SET_REMAINING_TIME);
export const setLeaderboardData = createAction(SET_LEADERBOARD_DATA);
export const setTeamScoreData = createAction(SET_TEAM_SCORE_DATA);
