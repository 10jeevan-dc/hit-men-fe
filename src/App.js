import React, { Component } from 'react';
import Proptypes from 'prop-types';
import { connect } from 'react-redux';
import Header from './components/Header/Header.component';
import StartPage from './components/StartPage/StartPage.component';
import GamePage from './components/GamePage/GamePage.component';
import LeaderBoardPage from './components/LeaderBoard/LeaderBoardPage.component';
import './App.css';

class App extends Component {
  getStartPage = () => (
    <div className="App">
      <Header />
      <StartPage />
    </div>
  );

  getGamePage = () => (
    <div className="App">
      <Header />
      <GamePage />
    </div>
  );

  getLeaderBoardPage = () => (
    <div className="App">
      <Header />
      <LeaderBoardPage />
    </div>
  );

  render() {
    const { screenID } = this.props;
    switch (screenID) {
      case 2: return this.getGamePage();
      case 3: return this.getLeaderBoardPage();
      default: return this.getStartPage();
    }
  }
}

const mapStateToProps = ({ game }) => ({
  screenID: game.gameState,
});

App.propTypes = {
  screenID: Proptypes.number,
};

App.defaultProps = {
  screenID: 1,
};

export default connect(mapStateToProps)(App);
