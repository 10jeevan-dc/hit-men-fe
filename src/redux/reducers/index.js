import { combineReducers } from 'redux';

import user from './user.reducer';
import game from './gameControl.reducer';

const combinedReducers = combineReducers({
  user,
  game,
});

export default combinedReducers;
