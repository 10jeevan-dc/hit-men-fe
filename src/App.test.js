import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import { combineReducers, createStore } from 'redux';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });

const mockReducers = {
  user: () => ({
    sessionName: '',
    userName: '',
    selfTeam: '',
    opponentTeam: '',
    gamePlayTimeInSeconds: 0,
    score: 0,
  }),
  game: () => ({
    gameState: 1,
    isUserAllowedToPlay: false,
    isUserDetailsSet: false,
    leaderBoardData: [],
    teamScoreData: {},
  }),
};

const mockStore = createStore(combineReducers(mockReducers), {});
it('renders without crashing for start page', () => {
  const wrapper = Enzyme.shallow(<App store={mockStore} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders without crashing for game page', () => {
  const wrapper = Enzyme.shallow(<App store={mockStore} />);
  wrapper.setProps({
    gameState: 2,
  });
  expect(wrapper).toMatchSnapshot();
});

it('renders without crashing for result page', () => {
  const wrapper = Enzyme.shallow(<App store={mockStore} />);
  wrapper.setProps({
    gameState: 3,
  });
  expect(wrapper).toMatchSnapshot();
});
