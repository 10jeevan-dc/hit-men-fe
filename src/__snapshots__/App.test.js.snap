// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing for game page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(App)
    gameState={2}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "gameState": 2,
      "screenID": 1,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "gameState": 2,
        "screenID": 1,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders without crashing for result page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(App)
    gameState={3}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "gameState": 3,
      "screenID": 1,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "gameState": 3,
        "screenID": 1,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders without crashing for start page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(App)
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "screenID": 1,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "screenID": 1,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
